sw/device/lib/dif/dif_otp_ctrl.h

```diff
@@ -0,0 +1,773 @@
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_OTP_CTRL_H_
+
+/**
+ * @file
+ * @brief <a href="/hw/ip/otp_ctrl/doc/">OTP</a> Device Interface Functions
+ */
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * A toggle state: enabled, or disabled.
+ *
+ * This enum may be used instead of a `bool` when describing an enabled/disabled
+ * state.
+ */
+typedef enum dif_otp_ctrl_toggle {
+  /*
+   * The "enabled" state.
+   */
+  kDifOtpCtrlToggleEnabled,
+  /**
+   * The "disabled" state.
+   */
+  kDifOtpCtrlToggleDisabled,
+} dif_otp_ctrl_toggle_t;
+
+/**
+ * A partition within OTP memory.
+ */
+typedef enum dif_otp_ctrl_partition {
+  /**
+   * The creator software configuration area.
+   *
+   * This partition contains device-specific calibration data.
+   */
+  kDifOtpCtrlPartitionCreatorSwCfg,
+  /**
+   * The owner software configuration area.
+   *
+   * This partition contains data to e.g. enable ROM hardening features.
+   */
+  kDifOtpCtrlPartitionOwnerSwCfg,
+  /**
+   * The hardware configuration area.
+   */
+  kDifOtpCtrlPartitionHwCfg,
+  /**
+   * The device lifecycle area.
+   */
+  kDifOtpCtrlPartitionLifeCycle,
+  /**
+   * Scrambled partition 0.
+   *
+   * This paritition contains TEST lifecycle state unlock tokens.
+   */
+  kDifOtpCtrlPartitionSecret0,
+  /**
+   * Scrambled partition 1.
+   *
+   * This partition contains SRAM and flash scrambling keys.
+   */
+  kDifOtpCtrlPartitionSecret1,
+  /**
+   * Scrambled partition 2.
+   *
+   * This partition contains the RMA unlock token and the CreatorRootKey.
+   */
+  kDifOtpCtrlPartitionSecret2,
+} dif_otp_ctrl_partition_t;
+
+/**
+ * Hardware instantiation parameters for OTP.
+ *
+ * This struct describes information about the underlying hardware that is
+ * not determined until the hardware design is used as part of a top-level
+ * design.
+ */
+typedef struct dif_otp_ctrl_params {
+  /**
+   * The base address for the OTP hardware registers.
+   */
+  mmio_region_t base_addr;
+} dif_otp_ctrl_params_t;
+
+/**
+ * Runtime configuration for OTP.
+ *
+ * This struct describes runtime information for one-time configuration of the
+ * hardware.
+ */
+typedef struct dif_otp_ctrl_config {
+  /**
+   * The number of bits in the pseudo-random integrity check period.
+   *
+   * For example, a value of 4 will ensure that the maximum number of cycles
+   * between checks is 0b1111 = 15.
+   *
+   * The maximum value is 31.
+   */
+  uint8_t integrity_period_bits;
+  /**
+   * The number of bits in the pseudo-random consistency check period.
+   *
+   * For example, a value of 4 will ensure that the maximum number of cycles
+   * between checks is 0b1111 = 15.
+   *
+   * The maximum value is 31.
+   */
+  uint8_t consistency_period_bits;
+} dif_otp_ctrl_config_t;
```

Comment:
The integrity check configuration CSRs have been modified in the meantime, and these period CSRs now just specify the mask directly (shifted by 8bits to the left in order to arrive at 40bit).
See https://github.com/lowRISC/opentitan/pull/3456.